### PR TITLE
Introduce wrapper contracts pallet In develop

### PR DESCRIPTION
**Pull request by satyamakgec**

### CODE REVIEW

**Review by adamdossa — APPROVED** [review]

File: pallets/contracts/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+        pub fn instantiate(
+            origin,
+            #[compact] endowment: BalanceOf<T>,
+            #[compact] gas_limit: Gas,
+            code_hash: CodeHash<T>,
+            data: Vec<u8>,
+            max_fee: BalanceOf<T>
+        ) -> DispatchResultWithPostInfo {
+            let sender = ensure_signed(origin.clone())?;
+            // Access the meta details of SE template
```
Comment (vkomenda) [inline pallets/contracts/src/lib.rs:244]: An additional check is possible:
```
pallet_permissions::Module<T>::ensure_call_permissions(&sender)?;
```

File: pallets/contracts/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+        /// Error type.
+        type Error = Error<T>;
+
+        /// The minimum amount required to generate a tombstone.
+        const NetworkShareInInstantiationFee: Perbill = T::NetworkShareInFee::get();
+
+        // Simply forwards to the `update_schedule` function in the Contract module.
+        #[weight = pallet_contracts::Call::<T>::update_schedule(schedule.clone()).get_dispatch_info().weight]
+        pub fn update_schedule(origin, schedule: Schedule) -> DispatchResult {
+            <pallet_contracts::Module<T>>::update_schedule(origin, schedule)
```
Comment (vkomenda) [inline pallets/contracts/src/lib.rs:173]: We can factor in
```
pallet_permissions::Module<T>::ensure_call_permissions(&sender)?;
```
if that makes sense. Not sure if you wanted a simple redirection or maybe we could gate it.

File: pallets/contracts/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+        // Simply forwards to the `call` function in the Contract module.
+        #[weight = pallet_contracts::Call::<T>::call(dest.clone(), *value, *gas_limit, data.clone()).get_dispatch_info().weight]
+        pub fn call(
+            origin,
+            dest: <T::Lookup as StaticLookup>::Source,
+            #[compact] value: BalanceOf<T>,
+            #[compact] gas_limit: Gas,
+            data: Vec<u8>
+        ) -> DispatchResultWithPostInfo {
+            <pallet_contracts::Module<T>>::call(origin, dest, value, gas_limit, data)
```
[the remark this answers is not in the document]
Comment (satyamakgec) [reply pallets/contracts/src/lib.rs:222]: We can skip this for this PR.. Even if we want we can add them in next task

File: pallets/staking/fuzzer/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+[package]
+name = "pallet-staking-fuzz"
```
[the remark this answers is not in the document]
Comment (satyamakgec) [reply pallets/staking/fuzzer/Cargo.toml:2]: I think it is a leftover because of merge conflicts.

**Review by vkomenda — APPROVED** [review]

**Pull request merged**